src/webapp/pages/form-page/disease-outbreak-event/utils/mapFormStateToEntityData.ts

```diff
@@ -0,0 +1,151 @@
+import {
+    DiseaseOutbreakEventBaseAttrs,
+    IncidentStatusType,
+} from "../../../../../domain/entities/disease-outbreak-event/DiseaseOutbreakEvent";
+import { DiseaseOutbreakEventWithOptions } from "../../../../../domain/entities/disease-outbreak-event/DiseaseOutbreakEventWithOptions";
+import { FormState } from "../../../../components/form/FormState";
+import { getHazardTypeValue } from "../../../../../data/repositories/consts/DiseaseOutbreakConstants";
+import { diseaseOutbreakEventFieldIds } from "./mapEntityToInitialFormState";
+import {
+    FormFieldState,
+    getAllFieldsFromSections,
+    getBooleanFieldValue,
+    getDateFieldValue,
+    getMultipleOptionsFieldValue,
+    getStringFieldValue,
+} from "../../../../components/form/FormFieldsState";
+
+export function mapFormStateToEntityData(
+    formState: FormState,
+    currentUserName: string,
+    diseaseOutbreakEventWithOptions: DiseaseOutbreakEventWithOptions
+): DiseaseOutbreakEventBaseAttrs {
+    const { diseaseOutbreakEvent } = diseaseOutbreakEventWithOptions;
+
+    const allFields: FormFieldState[] = getAllFieldsFromSections(formState.sections);
+
+    const hazardType = getStringFieldValue(diseaseOutbreakEventFieldIds.hazardType, allFields);
+
+    const diseaseOutbreakEventEditableData = {
+        name: getStringFieldValue(diseaseOutbreakEventFieldIds.name, allFields),
+        hazardType: getHazardTypeValue(hazardType),
+        mainSyndromeCode: getStringFieldValue(
+            diseaseOutbreakEventFieldIds.mainSyndromeCode,
+            allFields
+        ),
+        suspectedDiseaseCode: getStringFieldValue(
+            diseaseOutbreakEventFieldIds.suspectedDiseaseCode,
+            allFields
+        ),
+        notificationSourceCode: getStringFieldValue(
+            diseaseOutbreakEventFieldIds.notificationSourceCode,
+            allFields
+        ),
+        areasAffectedProvinceIds: getMultipleOptionsFieldValue(
+            diseaseOutbreakEventFieldIds.areasAffectedProvinceIds,
+            allFields
+        ),
+        areasAffectedDistrictIds: getMultipleOptionsFieldValue(
+            diseaseOutbreakEventFieldIds.areasAffectedDistrictIds,
+            allFields
+        ),
+        incidentStatus: getStringFieldValue(
+            diseaseOutbreakEventFieldIds.incidentStatus,
+            allFields
+        ) as IncidentStatusType,
+        emerged: {
+            date: getDateFieldValue(diseaseOutbreakEventFieldIds.emergedDate, allFields) as Date,
```

Comment:
That's definitely dangerous, the function says it returns `Date | null`

Reply:
Unsafe down-castings solved in PR https://github.com/EyeSeeTea/zebra-dev/pull/9